apps/onboarding/src/wallet/eip1167-proxy-deployer.ts

```diff
@@ -0,0 +1,74 @@
+import { networkConfig, NetworkConfig } from "@app/utils/config/network.config"
+import { eqAddress, Err, Ok, Result } from "@celo/base"
+import { ContractKit } from "@celo/contractkit"
+import { RawTransaction } from "@celo/contractkit/lib/wrappers/MetaTransactionWallet"
+import { Inject, Injectable, Scope } from "@nestjs/common"
+import { Contract} from "web3-eth-contract"
+import { abi as ProxyCloneFactoryABI } from '../../../../libs/blockchain/contracts/ProxyCloneFactory.json'
+import { WalletNotDeployed } from "./errors"
+import { ProxyDeployer } from "./wallet.service"
+
+/*
+ * XXX: This is a low-level implementation based on the contract version in 
+ * asaj/gas-efficiency. We should replace this with using a wrapper akin
+ * to how things look in the `legacy-proxy-deployer.ts`
+ */
+
+@Injectable({
+  // RelayerProxyService is Request scoped
+  scope: Scope.REQUEST
+})
+export class EIP1167ProxyDeployer implements ProxyDeployer {
```

Comment:
I guess thinking forward a bit, do you think we'd have a third `PooledEIP1167ProxyDeployer`?

Reply:
I think it mostly depends on whether we would need to fork functionality at that point. Currently we need to support both options because it introduces different behaviour in Valora, but if there's no difference between pooled and unpooled from Valora's point of view then I believe we can just upgrade this to the pooled variant without maintain both.